### PR TITLE
[ cleanup ] Reuse Tarjan's algorithm in JS backend and some other cleanups

### PR DESCRIPTION
**Pull request by stefan-hoeck**

### CODE REVIEW

**Comment by stefan-hoeck** [comment]
Please excuse me asking, but this is just a small change. Is something blocking its inclusion in master? I have some more JS related cleanups in the pipeline, so should I rather do them all in a single PR, or piecewise as I did so far?

**Comment by gallais** [comment]
@stefan-hoeck I am attending the Agda meeting this week so I haven't really
been looking at the Idris repo. I'll try to check this PR today.

**Comment by stefan-hoeck** [comment]
gallais, please, I didn't mean to rush you. Have fun (hopefully) at your meeting.

File: src/Compiler/ES/Imperative.idr [hunk 1]
```diff
@@ -410,7 +409,7 @@ compileToImperative c tm =
 
      -- list of toplevel definitions (only those necessary
      -- to implement the main expression)
-     lst_defs <- traverse getImp (defsUsedByNamedCExp ctm ndefs)
+     lst_defs <- traverse getImp ndefs
```
Comment (gallais) [inline src/Compiler/ES/Imperative.idr:412]: Has the deadcode been filtered out already?

**Comment by stefan-hoeck** [comment]
Rebased to master.

**Comment by gallais** [comment]
Wait I thought I had merged this already? I guess I was waiting for the CI to finish
and lost the tab after rebooting.

**Pull request merged**